core/src/main/java/com/neemre/btcdcli4j/core/client/BtcdClient.java

```diff
@@ -75,7 +75,31 @@ RedeemScript decodeScript(String hexRedeemScript) throws BitcoindException,
 
 	String encryptWallet(String passphrase) throws BitcoindException, CommunicationException;
 
-	BigDecimal estimateFee(Integer maxBlocks) throws BitcoindException, CommunicationException;
+	/**
+	 * See https://developer.bitcoin.org/reference/rpc/estimatesmartfee.html
+	 * @param maxBlocks Confirmation target in blocks (1 - 1008)
+	 * @param estimateMode optional, default=CONSERVATIVE
+	 *
+	 * @return
+	 * @throws BitcoindException
+	 * @throws CommunicationException
+	 */
+	BigDecimal estimateSmartFee(Integer maxBlocks, String estimateMode) throws BitcoindException, CommunicationException;
```

Comment:
Shouldn't estimateMode be an enum? Because it's a strict domain of values "Must be one of: “UNSET” “ECONOMICAL” “CONSERVATIVE”"

Reply:
Yeah that's better, udpated